accepted/0000-validate-signatures.md

```diff
@@ -0,0 +1,510 @@
+  valid `keyid`s, warning users to update the cli if a new key has been found
+  from npmjs.org.
+- 6. **Update the Keybase PGP key expiry**: We want to give folks with their own
+  tooling as much notice as possible while deprecating the use of the existing
+  PGP key. We will update the public Keybase key with a new expiration date (the
+  current key expires 2034-03-30). We will communicate this expiration to match
+  the deprecation timeframe (exact timing is TBD).
+- 7. **Stop generating PGP signatures once the key expires**: We will no longer
+  populate the packument field `dist.npm-signature` in new releases.
+
+#### Breaking changes
+
+Existing workflows validating PGP signatures will break once the public Keybase
+PGP expires and we stop writing this signature to packages `npm-signature`
+field.
+
+### Adding ECDSA signatures to packuments
+
+We will introduce a new `signatures` field in the version packument, e.g.
+[registry.npmjs.org/light-cycle/1.4.3](https://registry.npmjs.org/light-cycle/1.4.3):
+
+```
+{
+  "name":"light-cycle",
+  "description":"a consistent hash ring for your blue-glowing shards of PURE ENERGY",
+  "version":"1.4.3",
+  ...
+  "dist":{
+    "integrity":"sha512-sFcuivsDZ99fY0TbvuRC6CDXB8r/ylafjJAMnbSF0y4EMM1/1DtQo40G2WKz1rBbyiz4SLAc3Wa6yZyC4XSGOQ==",
+    "shasum":"c305f0113d81d880f846d84f80c7f3237f197bab",
+    "tarball":"https://registry.npmjs.org/light-cycle/-/light-cycle-1.4.3.tgz",
+    "fileCount":11,
+    "unpackedSize":25612,
+    "signatures": [{
+      "keyid": "SHA256:{{BASE64_SHA256_PUBLIC_KEY}}",
+      "sig": "a312b9c3cb4a1b693e8ebac5ee1ca9cc01f2661c14391917dcb111517f72370809..."
+    }],
+    "npm-signature":"-----BEGIN PGP SIGNATURE-----\r\nVersion: OpenPGP.js v3.0.4\r\nComment: https://openpgpjs.org\r\n\r\nwsFcBAEBCAAQBQJa0SiFCRA9TVsSAnZWagAAf40QAI6m9uc8N8iQ4xmVJI+y\naPICZ2wwSGY/LvjpSfMHcjiT9h92lijXQcF0bAdtUA6UPjI4e9GuzoBME52Q\nkfPPRQ06icX0Om8XJxNIeLgJ1PK9Odv5fmb4sOZZxk/4t1hhf15KdfJ7vQYr\nMbb6gI/QkcOgi8NHHJSAQDBxws679zj8f7j2qYr6RzaJ8v3kGlEDnjD4LxwU\nXkK9TALJmwrgocqQ05XO+k/C6sVAuM7dg3fuRJUQab6liyyxtfnIbNaam+2V\neDaf6IurFnkcwjfmtin9tq9pLE/Ml+MI38wqAmyNpWHCGPzJKZpd7nSPQQSR\n1M2n/yLPcTh7YBKgiDaGTdtjPgtq+gYiWzqXbQFw3ICLCDhwcemHMizuuhAu\nig8dsCleFnQW21eYCbP2s6H/Kp77NnYV2vFOLS2PobUapP9jVDYVOkfdGUKS\nYbhU3lhg7Dgx6WtHmvfmDyWClSljJir99rnlp8bxYzipVCEai+2SR371PwK5\nkkWNdd2dh2oIBnpZu6m/ksK+5Oz9Mq0cdpq8R2BSlUNAiRHjjDECCrHAxRs+\n276vyovxQlGhnuTKmbu4ivUD3i7TUp0RVmZHIjxt2+xFB99u1861MN20UFp6\nr6WmvstyORBFMHHlccT3a5y6mwQtMMId7ysc2hbn+FHURBClGmVv+frb72sk\n+GGk\r\n=AAks\r\n-----END PGP SIGNATURE-----\r\n"
+   }
+```
+
+The `keyid` will map to a public key hosted on `https://registry.npmjs.org/.well-known/npm-signature-keys`.
+
+## Rationale and Alternatives
+
+We propose a pragmatic improvement to the existing signatures and verification
+flow. We're not proposing any support for [signed
+packuments](https://github.com/npm/rfcs/pull/76) or user generated signatures
```

Comment:
I feel like we might want to expand this statement to accommodate our discussions around the notion that some form of "signed packument" **is** an appropriate mitigation for the "evil mirror" vector.

Reply:
Yup, moved this up under `Detailed explanation > Attack vector`